### PR TITLE
CANS-1014:  substance use warnings text updated to use dynamic item id for 7/8

### PR DESCRIPTION
**Pull request by LeonidMarushevskyi**
## JIRA
https://osi-cwds.atlassian.net/browse/CANS-1014

## Description

PR solves the issue with wrong substance use warnings text where after switching item 7 and 8 we are referencing the wrong id 8 for old assessments.
Added calculation of substance use items ids and propagation of this object to all components where we need to use substance use items ids. 
Implemented dynamic item Id for 8. SUBSTANCE USE item in assessment header, confidentiality warning after completion and on the print page.

## Tests

- [x] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [ ] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/Application/components/Assessment/AssessmentContainerInner.js [hunk 1]
```diff
@@ -170,6 +173,10 @@ AssessmentContainerInner.propTypes = {
   onAssessmentUpdate: PropTypes.func.isRequired,
   onCancelClick: PropTypes.func.isRequired,
   onEventDateFieldKeyUp: PropTypes.func.isRequired,
+  substanceUseItemsIds: PropTypes.shape({
+    underSix: PropTypes.array.isRequired,
```
Comment (zliuwd) [inline app/javascript/Application/components/Assessment/AssessmentContainerInner.js]: This propTypes block was used in couple different places, considering reusable approaching.

File: app/javascript/Application/components/Print/PrintAssessment.js [hunk 2]
```diff
@@ -145,8 +145,8 @@ class PrintAssessment extends PureComponent {
     return (
       <div style={alertSignBox}>
         <strong>
-          By selecting NO, Items 8, 48, and EC 41 (Substance Use Disorder Items) from this CANS assessment will be
-          redacted when printed.
+          By selecting NO, Items {this.props.substanceUseItemsIds.aboveSix[0]}, 48, and EC 41 (Substance Use Disorder
+          Items) from this CANS assessment will be redacted when printed.
```
Comment (zliuwd) [inline app/javascript/Application/components/Print/PrintAssessment.js]: Consider to separate it as a component.

Comment (LeonidMarushevskyi) [reply app/javascript/Application/components/Print/PrintAssessment.js]: PrintAssessment currently implemented as one component. I would postpone any refactoring until our print enhancement stories in next sprints.

File: app/javascript/Application/components/Assessment/AssessmentHelper.js [hunk 3]
```diff
@@ -244,6 +245,22 @@ export function isSubsequentType(type) {
   return AssessmentType.subsequent === type
 }
 
+export const getSubstanceUseItemsIds = assessment => {
+  const underSix = []
+  const aboveSix = []
+  assessment.state.domains.forEach(domain => {
```
Comment (nasar-cambria) [inline app/javascript/Application/components/Assessment/AssessmentHelper.js]: Did you wanted to just loop over the array here or check it's return value as well ?

Comment (LeonidMarushevskyi) [reply app/javascript/Application/components/Assessment/AssessmentHelper.js]: Loop over domains and items and find substance use items.

**Pull request merged**